errbot/__init__.py

```diff
@@ -49,7 +49,7 @@
 # Same happens with quotations marks, which are required for parsing
 # complex strings in arguments
 # Map of characters to sanitized equivalents
-ARG_BOTCMD_CHARACTER_REPLACEMENTS = {"—": "--", "“": '"', "”": '"', "’": "'", "‘": "'"}
+ARG_BOTCMD_CHARACTER_REPLACEMENTS = {"—": "--", "“": '"', "”": '"', "’": "'", "‘": "'", "\xa0": "\x20"}
```

Comment:
Why not have the replacement be a space, like so?

```suggestion
ARG_BOTCMD_CHARACTER_REPLACEMENTS = {"—": "--", "“": '"', "”": '"', "’": "'", "‘": "'", "\xa0": " "}
```

Reply:
@sijis  I have tried it with simply having a space, but in the bot logs it looked like this `--flag?argument` instead of `--flag argument`, when you specify the whitespace explicitly with a code, the logs display the second option. I made such decision being concerned for some unexpected behaviour, since I am not sure how unambiguous the " " is in Python.

Reply:
Ahh interesting.

Would you be able post a sample of the plugin code and the input parameters you've been using to test?

Reply:
@sijis I am not sure if I could post the code but basically, the bot would take a line such as `!jenkins build -b mybranch` and if for example the nbsp sneaks in between `-b` and `mybranch` it would fail, since the whole `-b\xa0mybranch` is going to be passed to the decorator @arg_botcmd, and the bot would just see it as an incorrect argument.